### PR TITLE
fnutils and errutils test cases, inline docs, listall fnc fix

### PR DESCRIPTION
**Pull request by neo7337**
1. added test cases for fnutils
2. added test cases for errutils
3. added inline docs for fnutils and errutils

### CODE REVIEW

File: fnutils/fn_test.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	//	_ = func() {
+	//		// This function should not be executed
+	//		t.Error("Function should not be executed")
+	//	}
+	//	timeout := 0 * time.Second
+	//
+	//	ExecuteAfter(nil, timeout)
+	//})
+	//
+	//t.Run("Execution does not occur when timeout is negative", func(t *testing.T) {
```
Comment (neo7337) [inline fnutils/fn_test.go:88]: @nandagopalan have a look at this condition if we need to cater for this in our logic

File: fnutils/fn_test.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+	//
+	//	go ExecuteAfter(fn, timeout)
+	//	time.Sleep(3 * time.Second)
+	//
+	//	if executionCount != 2 {
+	//		t.Errorf("Expected execution to occur twice, got %d times", executionCount)
+	//	}
+	//})
+	//
+	//t.Run("No execution occurs if the function is nil", func(t *testing.T) {
```
Comment (neo7337) [inline fnutils/fn_test.go:68]: @nandagopalan same here, if we need to handle this condition in our logic

Comment (nandagopalan) [reply fnutils/fn_test.go:68]: I think we may have to put a nil check in the code.

Comment (nandagopalan) [reply fnutils/fn_test.go:68]: I am approving this PR we will add that as a separate issue.

**Review by nandagopalan — APPROVED** [review]

**Pull request merged**